tests/test-leaks.lua

```diff
@@ -63,6 +62,12 @@ collectable(wibox.layout.align.horizontal())
 collectable(awful.widget.launcher({ image = cairo.ImageSurface(cairo.Format.ARGB32, 20, 20), command = "bash" }))
 collectable(awful.widget.prompt())
 collectable(awful.widget.textclock())
+
+-- Test the layout box. As its widgets are a singleton (per screen), clear them
+-- first.
+awful.widget.layoutbox.clear()
+collectgarbage("collect")
+collectgarbage("collect")
```

Comment:
Why the extra `collectgarbage` here? Also, does this really fix the leak? Apparently the old layoutbox is un-GC-able (perhaps something in the widget stuff keeps a reference to it?). This fixes the test by just creating a new layoutbox and ignoring that the previous one couldn't be collected. I'd rather figure out what references the layoutbox.

The only other test that does something with layoutboxes seems to be `test-benchmark.lua`, but it uses the same `create_wibox()` that this test already uses, so that shouldn't cause any leaks....